### DISCUSSION TITLE
GROUPING SET Re-Aggregation over different levels of detail

### DISCUSSION THREAD

**Discussion by Mytherin** [duckdb/duckdb#8769] [General]
Now that GROUPING SETS are supported we should add the optimization for different LoD. Rather than creating one aggregate HT per grouping set we should re-use the aggregate states from hash tables with many groups for the aggregates over few groups, essentially re-aggregating over another dimension.

For this the exported aggregate states would be very useful, since we could export the aggregate states from the early hash tables and re-combine/finalize them in the later ones. This way we should be able to implement this without requiring any changes to the hash tables themselves.

**Comment by glqstrauss** [comment]
I see that an issue referencing this issue was closed. Is the reuse optimization also implemented? We're experiencing memory issues running distinct counts with a large number of grouping sets (for understandable reasons) and trying to see if any internal optimizations may be available or upcoming prior to going forward with avoiding OOM by sharding our queries or switching to hyperloglog. Thanks!

**Comment by Mytherin** [comment]
This is not implemented yet.
